Sources/Baggage/BaggageContext.swift

```diff
@@ -98,17 +90,30 @@ public protocol BaggageContextProtocol {
     ///     }
     subscript<Key: BaggageContextKey>(_ key: Key.Type) -> Key.Value? { get set }
 
-    /// Iterates over the baggage context's contents invoking the callback one-by one.
+    /// Calls the given closure on each key/value pair in the `BaggageContext`.
     ///
-    /// - Parameter callback: invoked with the type erased key and value stored for the key in this baggage.
-    func forEach(_ callback: (AnyBaggageContextKey, Any) -> Void)
+    /// - Parameter body: A closure invoked with the type erased key and value stored for the key in this baggage.
+    func forEach(_ body: (AnyBaggageContextKey, Any) throws -> Void) rethrows
 }
 
 // ==== ------------------------------------------------------------------------
 // MARK: Baggage keys
 
-/// `BaggageContextKey`s are used as keys in a `BaggageContext`. Their associated type `Value` gurantees type-safety.
+/// `BaggageContextKey`s are used as keys in a `BaggageContext`. Their associated type `Value` guarantees type-safety.
 /// To give your `BaggageContextKey` an explicit name you may override the `name` property.
+///
+/// In general, `BaggageContextKey`s should be `internal` to the part of a system using it. If you want users of your code to be able
```

Comment:
I'm not sure whether we should recommend exposing things publicly in any way as it could lead people to put non-meta data values in the `BaggageContext` too. @ktoso WDYT?

Reply:
This needs a slight rewording I feel:

a) it is strongly recommended to do those extensions, using the keys directly is an anti pattern.
b) the example extension should be `var someID: ...` without public in front. People should not have to make those public unless some super blessed values -- i.e. `deadline` which we'll get to some day would most likley be a public thing.